src/services/localStorageManager.js

```diff
@@ -0,0 +1,54 @@
+export default {
+  setItem({ name, value }) {
+    let returnValue = null;
+
+    try {
+      localStorage.setItem(name, JSON.stringify(value));
+    } catch {
+      const valuesWithoutOldestData = this.removeOldestData();
+
+      if (valuesWithoutOldestData) {
+        returnValue = valuesWithoutOldestData;
+        this.setItem({ name, value });
+      }
+    }
+
+    return returnValue;
+  },
+
+  removeOldestData() {
+    const listsLog = JSON.parse(this.getItem('listsLog'));
```

Comment:
I think that getItem should also can parse and return already parsed data

Take into account that it could be an error of parsing and it should be handled